### PR TITLE
Added "Remove out of Office" and "Remove all shares"

### PR DESCRIPTION
**Pull request by panjkov**

### CODE REVIEW

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs [hunk 1]
```diff
@@ -46,10 +46,20 @@ public override List<ServiceObject> DescribeServiceObjects()
...
+            removeOutOfOffice.Validation.RequiredProperties.Add(Constants.SOProperties.OutOfOffice.DestinationUser);
+            removeOutOfOffice.ReturnProperties.Add(Constants.SOProperties.OutOfOffice.DestinationUser);
+            so.Methods.Add(removeOutOfOffice);
+
             Method listUserShares = Helper.CreateMethod(Constants.Methods.OutOfOfficeClient.ListUserShares, "Get all the destination users for OOF shares of current user", MethodType.List);
             listUserShares.ReturnProperties.Add(Constants.SOProperties.OutOfOffice.DestinationUser);
             so.Methods.Add(listUserShares);
 
+            Method removeAllShares = Helper.CreateMethod(Constants.Methods.OutOfOfficeClient.RemoveAllShares, "Remove All Shares", MethodType.Execute);
+            removeAllShares.ReturnProperties.Add(Constants.SOProperties.OutOfOffice.CallSuccess);
```
Comment (cyclops1982) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs]: I don't see this property being filled anywhere else in the change request.

I would suggest NOT to provide a 'success or not' output. If it doesn't work, just throw an exception.

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs [hunk 2]
```diff
@@ -266,5 +349,43 @@ private void ListSharedUsers()
...
+            using (Connection k2Con = new Connection())
+            {
+                k2Con.Open(base.K2ClientConnectionSetup);
+
+                // None for userstatus means the users is not configured, throw an exception
+                if (UserStatuses.None == k2Con.GetUserStatus())
+                    throw new ApplicationException(Constants.ErrorMessages.OutOfOfficeNotConfiguredForUser);
+
+                WorklistShares wsColl = k2Con.GetCurrentSharingSettings(ShareType.OOF);
+                if (wsColl != null)
```
Comment (cyclops1982) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs]: You can combine this IF statement:
if (wsColl != null && wsColl.Count > 0) {

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs [hunk 3]
```diff
@@ -228,6 +244,73 @@ private void AddOutOfOffice()
...
+
+                //  Throw error if multiple configurations (WorklistShare objects) detected, as this method cannot support that 
+                if (wsColl.Count != 1)
+                {
+                    throw new ApplicationException(Constants.ErrorMessages.MultipleOOFConfigurations);
+                }
+
+
+                //  If configuration exist already, remove user and re-share 
+                else if (wsColl.Count == 1)
```
Comment (cyclops1982) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs]: would this not be just an 'else' and not an else-if ?

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs [hunk 4]
```diff
@@ -266,5 +349,43 @@ private void ListSharedUsers()
...
+            serviceObject.Properties.InitResultTable();
+            DataTable results = base.ServiceBroker.ServicePackage.ResultTable;
+
+            using (Connection k2Con = new Connection())
+            {
+                k2Con.Open(base.K2ClientConnectionSetup);
+
+                // None for userstatus means the users is not configured, throw an exception
+                if (UserStatuses.None == k2Con.GetUserStatus())
+                    throw new ApplicationException(Constants.ErrorMessages.OutOfOfficeNotConfiguredForUser);
```
Comment (cyclops1982) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs]: I'd suggest {  and } for the IF-statement :)

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs (hunk 1 above)
Comment (panjkov) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs]: RemoveAllShares is that which is working, and Remove Out of Office is broken.

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs (hunk 3 above)
Comment (panjkov) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs]: It has to be done this way - the API allows to have multiple items in this collection, but the UI works properly only with first share in WSColl.

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs (hunk 4 above)
Comment (panjkov) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs]: Nuances :)

**Pull request merged**